doc/users/next_whats_new/separated_hatchcolor.rst

```diff
@@ -0,0 +1,64 @@
+Separated ``hatchcolor`` from ``edgecolor``
+-------------------------------------------
+
+The ``hatchcolor`` parameter for `~matplotlib.patches.Patch` objects has been
+separated from the ``edgecolor`` parameter. This allows for using different colors
+for hatches and edges by explicitly setting the ``hatchcolor`` parameter.
+
+Inherit Logic
+~~~~~~~~~~~~~
+When the *hatchcolor* parameter is specified, it will be used for the hatch.
+If it is not specified, it will fallback to using :rc:`hatch.color`.
+If this is a valid color, it will be used for the hatch.
+
+If the *hatch.color* rcParam is not set, then its default value will be used,
+which is *inherit*. In this case, *hatchcolor* will try to inherit from *edgecolor*
+of the patch if it is specified. If *edgecolor* is also not specified,
+*hatchcolor* will fallback to :rc:`patch.edgecolor`.
```

Comment:
```suggestion

The special value 'inherit' takes over the patch edgecolor, with a fallback to
:rc:`patch.edgecolor` if the patch edgecolor is 'none'.
```
Is this how it's implemented? - The logic/priority is: Synchronize hatchcolor with edgecolor, but if the patch does not have a visible edge, perfer to fall through to `rcParams['patch.edgecolor']` - we usually don't want the hatches to vanish.